app/org/apache/spark/deploy/history/SparkFSFetcher.scala

```diff
@@ -70,12 +69,7 @@ class SparkFSFetcher(fetcherConfData: FetcherConfigurationData) extends Elephant
    * properties in the configuration files. Triggering it too early will sometimes make the configuration object empty.
    */
   private lazy val _logDir: String = {
-    val conf = new Configuration()
```

Comment:
I am not totally confident about switching from webhdfs to hdfs. I would prefer retaining webhdfs. I think this will let Dr. Elephant read from hdfs even when installed outside the cluster, though we do not have concrete steps to do that yet.

Here, we also expect the namenode address to be configured in the FetcherConf.xml but how can it switch to the active namenode automatically?